src/main/drivers/bus_spi.c

```diff
@@ -546,7 +546,9 @@ void spiInitBusDMA()
                 bus->dmaTx = dmaGetDescriptorByIdentifier(dmaTxIdentifier);
                 bus->dmaTx->stream = DMA_DEVICE_INDEX(dmaTxIdentifier);
                 bus->dmaTx->channel = dmaTxChannelSpec->channel;
-                dmaInit(dmaTxIdentifier, OWNER_SPI_MOSI, device + 1);
+
```

Comment:
Should we roll this back if we fail to allocate both streams? See line 595, `// Disassociate channels from bus`. Note that any handling of this will be complicated somewhat by forthcoming changes in https://github.com/betaflight/betaflight/pull/10813 which support transmit only SPI DMAs. See https://github.com/SteveCEvans/betaflight/blob/f5fc4dd8e40e51a37365f62b5587b85ac0c56157/src/main/drivers/bus_spi.c#L604.

Reply:
I thought that SPI could operate in a 'TX DMA only' / 'RX DMA only' mode? If this assumption is wrong then yes, allocating both resources or bailing at the start of the initialisation will be better.

But since there is more change for SPI DMA coming, we could also wait for this until all the SPI DMA change is in.

Reply:
SPI will normally use both TX and RX (MOSI & MISO), but for some devices, notably the OSD, can make good use of just TX. The referenced PR adds that capability.